src/components/period-picker-with-period-type/PeriodPickerWithPeriodType.js

```diff
@@ -14,41 +14,49 @@ import AppContext from '../../pages/AppContext'
 import loadPeriodTypes from '../../redux/actions/periodTypes'
 import pluckPeriodTypes from '../../redux/selectors/periodTypes'
 
-/* i18n */
-
 /* styles */
 import styles from '../../utils/styles'
 
 export class PeriodPickerWithPeriodType extends PureComponent {
     state = {
-        periodTypes: [],
         selectedPeriodType: null,
         selected: null,
     }
 
-    /* FIXME: right now d2-ui periodPicker forces us to pass d2 through old  context api */
     getChildContext() {
         return {
             d2: this.props.d2,
         }
     }
 
     componentDidMount() {
-        const { loadPeriodTypes, periodTypes } = this.props
-        if (Array.isArray(periodTypes) && periodTypes.length === 0) {
-            loadPeriodTypes()
+        if (!this.props.periodTypes) {
+            this.props.loadPeriodTypes()
         }
     }
 
     onChangePeriodType = event => {
-        const selectedPeriodType = event.target.value
         this.setState({
-            selectedPeriodType,
+            selectedPeriodType: event.target.value,
         })
     }
 
     renderPeriodTypeDropdown = () => {
```

Comment:
This could be it's own component. Then we wouldn't need to call a method in the jsx of the render method and we can decrease the size of this component.